### PR TITLE
Update release checklist

### PR DESCRIPTION
**Pull request by siladu**
Takes into account latest changes from @cdivitotawela

### CODE REVIEW

**Review by usmansaleem — APPROVED** [review]
LGTM

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 1]
```diff
@@ -18,18 +18,12 @@ assignees: ''
...
-- [ ] Docker release startup test:
-    - `docker run hyperledger/besu:<version>`
-    - `docker run hyperledger/besu:<version>-arm64`
-    - `docker run --platform linux/amd64 hyperledger/besu:<version>-amd64`
-    - `docker run --pull=always hyperledger/besu:latest` (check version is <version>)
-- [ ] Merge homebrew PR
-- [ ] Publish Docs Release
+- [ ] Check binary SHAs are correct on the release page
+- [ ] Check "Container Verify" GitHub workflow has run successfully
+- [ ] Create homebrew PR - run https://github.com/hyperledger/homebrew-besu/blob/main/updateBesu.sh
```
Comment (macfarla) [inline .github/ISSUE_TEMPLATE/release-checklist.md:27]:
```suggestion
- [ ] Create homebrew release - run https://github.com/hyperledger/homebrew-besu/actions/workflows/update-version.yml
```

**Review by macfarla — APPROVED** [review]
there's a GHA for homebrew release now

**Comment by siladu** [comment]
On pause as there is an incoming change to besu-docs release workflow as well.

**Review by cdivitotawela — APPROVED** [review]
LGTM

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 2]
```diff
@@ -18,18 +18,13 @@ assignees: ''
...
-    - `docker run hyperledger/besu:<version>`
-    - `docker run hyperledger/besu:<version>-arm64`
-    - `docker run --platform linux/amd64 hyperledger/besu:<version>-amd64`
-    - `docker run --pull=always hyperledger/besu:latest` (check version is <version>)
-- [ ] Merge homebrew PR
-- [ ] Publish Docs Release
+- [ ] Check binary SHAs are correct on the release page
+- [ ] Check "Container Verify" GitHub workflow has run successfully
+- [ ] Create homebrew release - run https://github.com/hyperledger/homebrew-besu/actions/workflows/update-version.yml
+- [ ] Create besu-docs release - https://github.com/hyperledger/besu-docs/releases/new
```
Comment (macfarla) [inline .github/ISSUE_TEMPLATE/release-checklist.md:28]: do you want to add in here that you should copy the changelog from Besu

**Review by macfarla — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/release-checklist.md (hunk 2 above)
Comment (siladu) [reply .github/ISSUE_TEMPLATE/release-checklist.md:28]: done

**Pull request merged**